casdk-docs/blog/2024-05-16-dotnet8/index.md

```diff
@@ -0,0 +1,156 @@
+
+* Change the version of container images to build and test
+* Update the `.csproj` file
+* Replace `Startup.cs` with `Program.cs`
+* Replace `FunctionName` attribute with `Function`
+* Change loggers to be obtained from the DI container
+* Update the document
+
+For more details browse: [Pull Request #420](https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/420).
+
+## Port Number Breaking change
+
+As Carbon Aware SDK WebAPI uses ASP.NET Core technology another collateral must do change was required since .NET 8 changed its default port number from 80 to 8080 [Microsoft Learn document](https://learn.microsoft.com/en-us/dotnet/core/compatibility/containers/8.0/aspnet-port)).
+
+Changing the port number from WebAPI container, which affects the containerPort in Helm chart and some GitHub Workflows which uses WebAPI.
+
+## Broken build pipeline on GitHub Actions
+
+Thanks to GitHub, a lot of automation is available in order to publish code, allowing to focus more on coding, in particular the Carbon Aware SDK repository is configured to publish WebAPI container image (like a snapshot build) when a commit occurs on the dev branch. 
+
+However, it suddenly stopped working after .NET 8 upgrade.
+
+The team investigated the logs (Fig. 2), as a container image for both AMD64 and Arm64 Linux in GitHub Actions with [docker/build-push-action](https://github.com/docker/build-push-action): a mysterious segmentation fault (SEGV) was occurring after the upgrade… the code was not changed,  `dotnet publish` was outside the scope.
+
+```
+ > [linux/arm64 build-env 4/6] RUN dotnet publish CarbonAware.WebApi/src/CarbonAware.WebApi.csproj -c Release -o publish:
+7.209 MSBuild version 17.9.6+a4ecab324 for .NET
+24.69   Determining projects to restore...
+41.42 Segmentation fault (core dumped)
+```
+
+Fig.2 Logs in `dotnet publish` on GitHub Actions
+
+Further investigation was done, and thanks to a [.NET blog](https://devblogs.microsoft.com/dotnet/improving-multiplatform-container-support/), about multi-platform container support, that an unsupported approach was used for the build, and needed to be amended. More precisely, since .NET 6, QEMU static binaries were used to build container image for multi platforms.
+
+Fortunately .NET blog guides how to build multi platform container images, and the workflow was fixed accordingly in [Pull Request #498](https://github.com/Green-Software-Foundation/carbon-aware-sdk/pull/498). So WebAPI container image with .NET 8 can be pulled from [GitHub Packages](https://github.com/Green-Software-Foundation/carbon-aware-sdk/pkgs/container/carbon-aware-sdk) now!
+
+# Use case in NTT / NTT DATA
+
+While NTT & NTT DATA have been contributing to the Carbon Aware SDK a long time, none appeared in the [adopters list](https://github.com/Green-Software-Foundation/carbon-aware-sdk/blob/dev/casdk-docs/docs/overview/adopters.md), it is now changed thanks to those uses cases.
```

Comment:
…a long time, we had not publicly referenced our adoption of the tool. 
The Carbon Aware SDK v1.4.0 carbon metrics exporter (dependent on .NET8 upgrade) has increased the usefulness of the SDK due to increased visualization capabilities. …